crates/cli/src/main.rs

```diff
@@ -275,6 +282,93 @@ mod linux {
     }
 }
 
+#[cfg(target_os = "linux")]
+mod flatpak {
+    use std::ffi::OsString;
+    use std::path::PathBuf;
+    use std::process::Command;
+    use std::{env, process};
+
+    const EXTRA_LIB_ENV_NAME: &'static str = "ZED_FLATPAK_LIB_PATH";
+
+    /// Adds bundled libraries to LD_LIBRARY_PATH if running under flatpak
+    pub fn ld_extra_libs() {
+        let mut paths = if let Ok(paths) = env::var("LD_LIBRARY_PATH") {
+            env::split_paths(&paths).collect()
+        } else {
+            Vec::new()
+        };
+
+        if let Ok(extra_path) = env::var(EXTRA_LIB_ENV_NAME) {
+            paths.push(extra_path.into());
+        }
+
+        env::set_var("LD_LIBRARY_PATH", env::join_paths(paths).unwrap());
+    }
+
+    /// Restarts outside of the sandbox if currently running within it
+    pub fn try_restart_to_host() {
+        if let Some(flatpak_dir) = get_flatpak_dir() {
```

Comment:
It appears the current behavior is to always trigger the sandbox bypass if the `FLATPAK_ID` environment variable is detected, and moreover crash Zed if it fails. Could we have a way to opt out of it (e.g. if it consistently fails to work, or to use Zed within another app's sandbox)? Perhaps an extra environment variable such as `ZED_FLATPAK_BYPASS=0` would work, and could be mentioned in the error message as a way around the failure. A graceful failure (which just attempts to start Zed anyway) or gating it behind a check for `FLATPAK_ID == dev.zed.Zed*` are potential alternatives if that's not desired, though a separate env var seems more flexible, in principle.

Reply:
This should've fixed it. I guess the problem was that if the binary was ran from a different Flatpak it would get the wrong env variable and then restart and look for the app binary there, which obviously wouldn't exist, thus causing a crash. Now it just checks that the flatpak id is indeed correct and if it isn't then it starts without flatpak spawn.

Reply:
Thank you, that will help. Although I still recommend adding a escape hatch so users which cannot use the sandbox escape due to compatibility issues (e.g. NixOS users) have at least a workaround they can resort to (I guess it'd be frustrating to tell them they just can't use the official Zed Flatpak at all).

Reply:
I've added an environment variable, `ZED_FLATPAK_NO_ESCAPE` which when set will skip the sandbox escape. I've also added the relevant permissions to the manifest such that it isn't totally useless when not escaping (but plenty of things are still broken if you use it).

Reply:
Thank you very much! That's good enough for me. Good job!

Regarding permissions and stuff, we may want to take a look at how the VSCode Flatpak does it, as I imagine the permissions would be similar. But we can always test and add more over time, of course!

(By the way, you forgot a debug print at line 349 of `main.rs`.)

Reply:
Whoops, and another one at 353.